File: Serene/Serene.Script/Northwind/Employee/EmployeeFormatter.cs

namespace Serene.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System.Collections.Generic;

    public class EmployeeFormatter : ISlickFormatter, IInitializeColumn
    {
        public string Format(SlickFormatterContext ctx)
        {
            var text = Q.HtmlEncode(ctx.Value);

            if (string.IsNullOrEmpty(GenderProperty))
                return text;

            var gender = ctx.Item[GenderProperty] as Gender?;
            
            return "<span class='" + 
                (gender == Gender.Female ? "employee-symbol female" : "employee-symbol male") + 
                "'>" + text + "</span>";
        }

        [Option]
        public string GenderProperty { get; set; }

        public void InitializeColumn(SlickColumn column)
        {
            column.ReferencedFields = column.ReferencedFields ?? new List<string>();

            if (!string.IsNullOrEmpty(GenderProperty))
            {
                column.ReferencedFields.Add(GenderProperty);
                return;
            }
        }
    }
}

File: Serene/Serene.AspNetCore/Modules/Northwind/Employee/EmployeeFormatter.ts
namespace Serene.Northwind {

    @Serenity.Decorators.registerFormatter([Serenity.ISlickFormatter, Serenity.IInitializeColumn])
    export class EmployeeFormatter implements Slick.Formatter {
        format(ctx: Slick.FormatterContext) {
            var text = Q.htmlEncode(ctx.value);

            if (!this.genderProperty) {
                return text;
            }

            var gender = ctx.item[this.genderProperty];
            return "<span class='" + ((gender === Gender.Female) ?
                'employee-symbol female' : 'employee-symbol male') +
                "'>" + text + '</span>';
        }

        @Serenity.Decorators.option()
        public genderProperty: string;

        public initializeColumn(column: Slick.Column) {
            column.referencedFields = column.referencedFields || [];
            if (this.genderProperty)
                column.referencedFields.push(this.genderProperty);
        }
    }
}